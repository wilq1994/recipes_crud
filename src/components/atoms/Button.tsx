import styled, { css } from "styled-components";

const Button = styled.button<{ small?: boolean; secondary?: boolean }>`
  background: transparent;
  color: var(--primary-color);
  font-weight: 700;
  font-size: 1rem;
  padding: 1em 1.5em;
  margin: 0;
  border: 0;
  outline: 0;
  border-radius: 3px;
  box-shadow: 0 0 0 1px var(--primary-color) inset;
  transition: background 0.3s ease, box-shadow 0.3s ease;
  cursor: pointer;
  min-width: 120px;
  &:hover {
    background: #fff;
    box-shadow: 0 0 0 3px var(--primary-color) inset;
  }

  ${(props) =>
    props.small &&
    css`
      padding: 0.5em 1em;
      min-width: auto;
    `}

  ${(props) =>
    props.secondary &&
    css`
        color: var(--tertiary-color);
        box-shadow: 0 0 0 1px var(--tertiary-color) inset;
        &:hover {
            box-shadow: 0 0 0 3px var(--tertiary-color) inset;
        }
    `}
`;

export default Button;
