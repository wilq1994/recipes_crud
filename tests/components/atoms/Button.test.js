import React from "react";
import { shallow } from "enzyme";

import Button from "components/atoms/Button";

describe("Button", () => {
  it("should render and match snapshot", () => {
    const component = shallow(<Button>Button</Button>);

    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it("should render small if small property passed", () => {
    const component = shallow(<Button small>Button</Button>);

    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it("should render as secondary if secondary property passed", () => {
    const component = shallow(<Button secondary>Button</Button>);

    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
