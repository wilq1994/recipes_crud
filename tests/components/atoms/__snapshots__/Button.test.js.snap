// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and match snapshot 1`] = `
<button
  className="sc-bdnylx hCJbVq"
>
  Button
</button>
`;

exports[`Button should render as secondary if secondary property passed 1`] = `
<button
  className="sc-bdnylx kPwPpY"
>
  Button
</button>
`;

exports[`Button should render small if small property passed 1`] = `
<button
  className="sc-bdnylx kBzZJx"
>
  Button
</button>
`;
